app/controllers/provider/sessions_controller.rb

```diff
@@ -101,4 +100,9 @@ def session_return_to
   def instantiate_sessions_presenter
     @presenter = Provider::SessionsPresenter.new(domain_account)
   end
+
+  def redirect_to_enforced_sso
+    return if !domain_account.settings.enforce_sso? || published_authentication_providers.count != 1
```

Comment:
```suggestion
    return unless domain_account.settings.enforce_sso? && published_authentication_providers.count == 1
```

Reply:
Then it can also become:
```suggestion
    return unless domain_account.settings.enforce_sso? && published_authentication_providers.one?
```